src/main/java/Duke.java

```diff
@@ -1,10 +1,73 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        Ui.greeting();
+
+        Scanner sc = new Scanner(System.in);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/MM/yyyy HHmm");
+        TaskList taskList = new TaskList();
+
+        Parser parser = new Parser(sc.nextLine());
+        while (parser.isCommandNotBye()) {
+            try {
+                Command taskType = parser.getCommand();
+                switch (taskType) {
+                    case LIST:
+                        Ui.execute(Command.LIST, taskList, null);
+                        break;
+                    case DELETE:
+                        Task deletedTask = taskList.get(parser.getTaskIndex() - 1);
+                        taskList.delete(parser.getTaskIndex());
+                        Ui.execute(Command.DELETE, taskList, deletedTask.getStatus());
+                        break;
+                    case DONE:
+                        Task doneTask = taskList.get(parser.getTaskIndex() - 1);
+                        doneTask.setDone(true);
+                        Ui.execute(Command.DONE, taskList, doneTask.getStatus());
+                        break;
+                    case DEADLINE:
+                    case EVENT:
+                    case TODO:
+                        if (parser.hasDescriptionError()) {
```

Comment:
Perhaps you could have a helper function to avoid deep nesting for better readability :)

https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting